### PR TITLE
Revert "gnrc_netreg: some optimisations"

### PR DESCRIPTION
**Pull request by miri64**
Reverts RIOT-OS/RIOT#8517

See https://github.com/RIOT-OS/RIOT/pull/8517#discussion_r169728389

### CODE REVIEW

File: sys/net/gnrc/application_layer/tftp/gnrc_tftp.c [hunk 1]
```diff
@@ -367,7 +367,7 @@ int _tftp_init_ctxt(ipv6_addr_t *addr, const char *file_name,
     /* generate a random source UDP source port */
     do {
         ctxt->src_port = (random_uint32() & 0xff) + GNRC_TFTP_DEFAULT_SRC_PORT;
-    } while (gnrc_netreg_lookup(GNRC_NETTYPE_UDP, ctxt->src_port));
+    } while (gnrc_netreg_num(GNRC_NETTYPE_UDP, ctxt->src_port));
```
Comment (smlng) [inline sys/net/gnrc/application_layer/tftp/gnrc_tftp.c]: reverting and reintroducing this is suboptimal

File: sys/net/gnrc/network_layer/ipv6/gnrc_ipv6.c [hunk 2]
```diff
@@ -219,7 +219,7 @@ static void _dispatch_next_header(gnrc_pktsnip_t *current, gnrc_pktsnip_t *pkt,
 
     /* dispatch IPv6 extension header only once */
     if (should_dispatch_current_type) {
-        bool should_release = (!gnrc_netreg_lookup(GNRC_NETTYPE_IPV6, nh)) &&
```
Comment (smlng) [inline sys/net/gnrc/network_layer/ipv6/gnrc_ipv6.c]: same here

**Comment by miri64** [comment]
> I would not revert this completely, but rather (re)add the gnrc_netreg_num function using _netreg_lookup internally to iterate over the array and count entries.

Good thing we split up the PR in several commits back then ;-).

**Comment by miri64** [comment]
Closing and deleting the branch in this one, though since revert branches are opened in RIOT-OS/RIOT and I don't want to force push in mainline.

**Comment by miri64** [comment]
@smlng See #8622.

**Pull request closed without merging**